### PR TITLE
Add support for invalidation timestamp in QueryApiKey

### PR DESCRIPTION
**Pull request by jfreden**
This is a follow up PR from https://github.com/elastic/elasticsearch/pull/102472. This adds the ability to use `invalidation` timestamp as a valid [query value](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-query-api-key.html#security-api-query-api-key-request-body) in the QueryApiKey API.

### CODE REVIEW

**Review by n1v0lg — APPROVED** [review]
LGTM! 

Let's figure out why GH insists on adding the `external-contributor` label to your PRs, next week 👀

File: x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryApiKeyIT.java [hunk 1]
```diff
@@ -387,6 +387,12 @@ public void testExistsQuery() throws IOException, InterruptedException {
         // Create an invalidated API key
         createAndInvalidateApiKey("test-exists-4", authHeader);
 
+        // Get the invalidated API key
+        assertQuery(authHeader, """
+            {"query": {"exists": {"field": "invalidation" }}}""", apiKeys -> {
```
[the remark this answers is not in the document]
Comment (jfreden) [reply x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryApiKeyIT.java:392]: Good idea! I've added it to the `testQuery` method.

File: docs/reference/rest-api/security/get-api-keys.asciidoc [hunk 2]
```diff
@@ -175,7 +175,7 @@ A successful call returns a JSON structure that contains the information of the
 <4> Creation time for the API key in milliseconds
 <5> Optional expiration time for the API key in milliseconds
 <6> Invalidation status for the API key. If the key has been invalidated, it has
-a value of `true`. Otherwise, it is `false`.
+a value of `true` and an additional field with the `invalidation` time in milliseconds. Otherwise, it is `false`.
```
[the remark this answers is not in the document]
Comment (jfreden) [reply docs/reference/rest-api/security/get-api-keys.asciidoc:178]: Yes, good point. I added it to the sample initially but it feels like invalidation is not what someone reading the get api key docs is looking for (probably want the regular "happy path") and adding a separate sample for the purpose of `invalidation` felt overkill, so I decided to leave it out.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyBoolQueryBuilder.java [hunk 3]
```diff
@@ -37,6 +37,7 @@ public class ApiKeyBoolQueryBuilder extends BoolQueryBuilder {
         "doc_type",
         "name",
         "api_key_invalidated",
+        "invalidation_time",
```
[the remark this answers is not in the document]
Comment (jfreden) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyBoolQueryBuilder.java:40]: Nice! Added.

File: docs/reference/rest-api/security/query-api-key.asciidoc [hunk 4]
```diff
@@ -77,6 +77,9 @@ Expiration time of the API key in milliseconds.
 Indicates whether the API key is invalidated. If `true`, the key is invalidated.
 Defaults to `false`.
 
+`invalidation`::
+Invalidation time of the API key in milliseconds.
```
[the remark this answers is not in the document]
Comment (jfreden) [reply docs/reference/rest-api/security/query-api-key.asciidoc:81]: Good update!

**Pull request merged**